wsagent/che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitConnection.java

```diff
@@ -730,44 +740,144 @@ public void init(InitRequest request) throws GitException {
         }
     }
 
+    /** @see org.eclipse.che.api.git.GitConnection#log(org.eclipse.che.api.git.shared.LogRequest) */
     @Override
-    public LogPage log(LogRequest request) throws GitException {
+    public LogPage log(LogRequest logRequest) throws GitException {
         LogCommand logCommand = getGit().log();
+        String filePath = null;
         try {
-            setRevisionRange(logCommand, request);
-
-            request.getFileFilter().forEach(logCommand::addPath);
-
+            setRevisionRange(logCommand, logRequest);
+            setPaging(logCommand, logRequest);
+            if (logRequest != null) {
+                logRequest.getFileFilter().forEach(logCommand::addPath);
+                filePath = logRequest.getFilePath();
+                if (!isNullOrEmpty(filePath)) {
+                    logCommand.addPath(filePath);
+                }
+            }
             Iterator<RevCommit> revIterator = logCommand.call().iterator();
             List<Revision> commits = new ArrayList<>();
-
             while (revIterator.hasNext()) {
                 RevCommit commit = revIterator.next();
-                PersonIdent committerIdentity = commit.getCommitterIdent();
-
-                GitUser gitUser = newDto(GitUser.class).withName(committerIdentity.getName())
-                                                       .withEmail(committerIdentity.getEmailAddress());
-
-                Revision revision = newDto(Revision.class).withId(commit.getId().getName())
-                                                          .withMessage(commit.getFullMessage())
-                                                          .withCommitTime(MILLISECONDS.convert(commit.getCommitTime(), SECONDS))
-                                                          .withCommitter(gitUser);
+                Revision revision = getRevision(commit, filePath);
                 commits.add(revision);
             }
             return new LogPage(commits);
         } catch (GitAPIException | IOException exception) {
             String errorMessage = exception.getMessage();
+            LOG.error("Failed to retrieve log. ", exception);
```

Comment:
I think this logging error should be placed at line 772, because this exception can be thrown when calling log but no commits present